.pipelines/.vsts-vhd-automation.yaml

```diff
@@ -5,61 +5,146 @@ pool:
 
 parameters:
 - name: ImageBump
-  displayName: Image Bump + Branch Cutting
+  displayName: Image Bump + Create Official Branch
   type: boolean
   default: true
 - name: ReleaseNotes
-  displayName: Release Notes
+  displayName: Generate Release Notes
   type: boolean
   default: true
-- name: ArtifactTrigger
-  displayName: SIG Artifact Trigger
+- name: CreateRelease
+  displayName: Create SIG Release
   type: boolean
   default: true
 
 variables:
   - group: "AKS Dev Assistant (KV)"
 
-steps:
-- bash: |
-        set -x
-        echo "Removing existing go environment"
-        sudo rm -r /usr/local/go
-        #GOLANG_VERSION=$(curl -s 'https://go.dev/VERSION?m=text' | head -n 1)
-        GOLANG_VERSION="1.20.6"
-        echo "Downloading ${GOLANG_VERSION}"
-        curl -O "https://dl.google.com/go/${GOLANG_VERSION}.linux-amd64.tar.gz"
+stages:
+  - stage: bump_image_version
+    dependsOn: []
+    condition: eq('${{ parameters.ImageBump }}', true)
+    jobs:
+    - job: automation
+      timeoutInMinutes: 15
+      steps:
+      - bash: |
+          az extension add -n azure-devops
+          echo $MAPPED_ADO_PAT | az devops login --organization=https://dev.azure.com/msazure
+          az devops configure --defaults organization=https://dev.azure.com/msazure project=CloudNativeCompute
+        env:
+          MAPPED_ADO_PAT: $(PAT-aksdevassistant)
+        displayName: 'az devops login'
+      - bash: |
+          echo "PR for Image Bumping, Official Branch Cutting"
+          /bin/bash vhdbuilder/scripts/automate_version_bump.sh $(IMAGE_VERSION) $MAPPED_GITHUB_PAT "${BUILD_ID[@]}"
+        env:
+          MAPPED_GITHUB_PAT: $(GITHUB_PAT)
+        displayName: 'bump image version and create official branch'
+  - stage: generate_release_notes
+    dependsOn: []
+    condition: eq('${{ parameters.ReleaseNotes }}', true)
+    jobs:
+    - job: automation
+      timeoutInMinutes: 30
+      steps:
+      - bash: |
+          set -x
+          echo "Removing existing go environment"
+          sudo rm -r /usr/local/go
+          GOLANG_VERSION="1.22.2"
+          echo "Downloading ${GOLANG_VERSION}"
+          curl -O "https://dl.google.com/go/${GOLANG_VERSION}.linux-amd64.tar.gz"
```

Comment:
There was a security item this sprint which forbidden access to the internet during build pipelines. 

1-  can we prebake this in the agent ? 
2- is there an hosted version on MAR ?

Reply:
let me look into this, I can't actually remember the reason why we ended up needing to do adhoc setup like we're doing now

Reply:
I think we should consider using a 1ES-managed custom image - that way we can specify the dependencies we want, while also getting automatic updates

right now I believe we're using a static 2004 base image

Reply:
we also might want to consider switching over to azure linux as well

Reply:
so I was wrong, we are actually using a 2004 image that gets updated by CloudTest, just verified that the image already has the correct go version we need, and thus we're able to get rid of the step altogether